qcodes/utils/dataset/doNd.py

```diff
@@ -283,11 +282,13 @@ def do1d(
 
         for set_point in tqdm(setpoints, disable=not show_progress):
             param_set.set(set_point)
+            t0 = time.time()
             datasaver.add_result(
                 (param_set, set_point), *call_param_meas(), *additional_setpoints_data
             )
-
-    param_set.post_delay = original_delay
+            datasaver_time = abs(time.time() - t0)
```

Comment:
I suggest using time.perf_counter rather than time.time since that has a better resolution https://docs.python.org/3/library/time.html#time.perf_counter